hibernate-core/src/main/java/org/hibernate/persister/collection/AbstractCollectionPersister.java

```diff
@@ -300,6 +313,19 @@ public AbstractCollectionPersister(
 			spaces[i] = tables.next();
 		}
 
+		/**
+		 * Add the predicate on the role in the WHERE clause before creating the SQL queries.
+		 */
+		BasicValue discriminatorDescriptor = collectionBootDescriptor.getMappedByAnyDiscriminatorDescriptor();
```

Comment:
Instead of introducing `getMappedByAnyDiscriminatorDescriptor`, how about you resolve this `Value` here based on `mappedBy`?

Reply:
If I'm not mistaken, I have all the information in `CollectionBinder `regarding the discriminator column, and I no longer have this information, or any information that would allow me to find the discriminator column, in `AbstractCollectionPersister`. 

That's why I use a pass-through with `org.hibernate.mapping.Collection`

But perhaps I have information that I haven't seen.

![image](https://github.com/hibernate/hibernate-orm/assets/1230949/c8778578-ed7a-47e1-956f-a0b1c75807e4)

I have lost the information from the referenced class and I only have the name of the return variable attribute.